### PR TITLE
Fix: "Error getting handoff location" Toast 

### LINKED ISSUE #163: Handoff location Bug
Repro:
- Go to a Non-Accepted state book then go to an Accepted state book
- a Toast notification will pop up saying "failed to get handoff location"

### PR DESCRIPTION
**Pull request by cyrusdiego**
# Summary
- Refactor to listen to the Requests collection for the currently selected book 
- To Reproduce the bug:
    - Accept a request on a book and select a handoff location. 
    - Navigate back to Library and select any other book (I found the bug occurred when selecting a "Requested" book)
    - Navigate to a previously accepted book and a toast notification will pop up.
    - Note: I think Caleb also saw this bug just from accepting a book 

Closes #163

### CODE REVIEW

**Review by ghassanzadeh — APPROVED** [review]
LGTM!

File: app/src/main/java/com/example/unlibrary/library/LibraryRepository.java [hunk 1]
```diff
@@ -106,6 +107,44 @@ public LibraryRepository(FirebaseFirestore db, FirebaseAuth auth, Client algolia
...
+                    if (error != null) {
+                        Log.e(TAG, "Error getting handoff location", error);
+                        return;
+                    }
+
+                    // If the request is not in either Accepted or Borrowed state, location is null
+                    // null location is handled in viewmodel / view (map is not initiated and location
+                    // pointer is not accessed)
+                    List<Request> requests = value.toObjects(Request.class);
+                    listener.onSuccess(requests.get(0).getLocation());
```
Comment (calebschoepp) [inline app/src/main/java/com/example/unlibrary/library/LibraryRepository.java:135]: Can this fail? i.e. the requests doesn't have a 0th element?

**Review by calebschoepp — COMMENTED** [review]
Good work. Just one question about a possible failure in the listener.

**Comment by calebschoepp** [comment]
LGTM

**Review by calebschoepp — APPROVED** [review]

**Pull request merged**